frontend/src/component/feature/FeatureStrategy/FeatureStrategyConstraints/FeatureStrategyConstraints.tsx

```diff
@@ -16,12 +16,37 @@ interface IFeatureStrategyConstraintsProps {
     >;
 }
 
+const filterConstraints = (constraint: any) => {
+    if (constraint.hasOwnProperty('values')) {
+        return constraint.values && constraint.values.length > 0;
+    }
+
+    if (constraint.hasOwnProperty('value')) {
+        return constraint.value !== '';
+    }
+};
+
 export const FeatureStrategyConstraints = ({
     projectId,
     environmentId,
     strategy,
     setStrategy,
 }: IFeatureStrategyConstraintsProps) => {
+    useEffect(() => {
+        return () => {
+            if (!strategy.constraints) {
+                return;
+            }
+
+            // If the component is unmounting we want to remove all constraints that do not have valid single value or
+            // valid multivalues
+            setStrategy((prev) => ({
+                ...prev,
+                constraints: prev.constraints?.filter(filterConstraints),
+            }));
+        };
+    }, []);
```

Comment:
If the component unmounts, we look at the strategy constraints and filter them by value or values, depending on the constraint object. If they don't have any values, we remove them.